subsys/random/rand32_timer.c

```diff
@@ -14,21 +14,47 @@
  * provide a random number generator.
  */
 
+#include <zephyr/init.h>
 #include <zephyr/random/rand32.h>
 #include <zephyr/drivers/timer/system_timer.h>
 #include <zephyr/kernel.h>
-#include <zephyr/sys/atomic.h>
 #include <string.h>
+#include <zephyr/device.h>
 
 #if defined(__GNUC__)
 
-/*
- * Symbols used to ensure a rapid series of calls to random number generator
- * return different values.
- */
-static atomic_val_t _rand32_counter;
 
-#define _RAND32_INC 1000000003U
+static uint32_t state[4];
+
+static inline uint32_t rotl(const uint32_t x, int k)
+{
+	return (x << k) | (x >> (32 - k));
+}
+
+static int xoshiro128_init_state(const struct device *dev)
```

Comment:
why not just initialize it when you declare it?  What's the point of this function?

Reply:
that was an artifact from when I was trying to do something else to get the initial state values, I have now changed it